### ISSUE TITLE
Developing locally plugin packages throws wrong error when user is not a global owner of npm

### ISSUE THREAD

**Issue by masives** [netlify/cli#1029]
**- Do you want to request a _feature_ or report a _bug_?**
bug
**- What is the current behavior?**
When I run `netlify build --dry` I receive error below. However this error doesn't occur when I run command with sudo.
```

❯ Installing plugins dependencies

┌─────────────────────────────┐
│    Netlify Build Error!     │
└─────────────────────────────┘

TypeError: 'undefined' plugin not installed or found
Path must be a string.
    at resolve (/home/jozef-piecyk-10clouds/.config/yarn/global/node_modules/resolve/lib/async.js:47:19)
    at Promise (internal/util.js:274:30)
    at new Promise (<anonymous>)
    at resolve (internal/util.js:273:12)
    at resolvePlugin (/home/jozef-piecyk-10clouds/.config/yarn/global/node_modules/@netlify/build/src/plugins/install.js:28:30)
    at Promise.all.pluginsOptions.map.pluginOptions (/home/jozef-piecyk-10clouds/.config/yarn/global/node_modules/@netlify/build/src/plugins/install.js:17:81)
    at Array.map (<anonymous>)
    at installPlugins (/home/jozef-piecyk-10clouds/.config/yarn/global/node_modules/@netlify/build/src/plugins/install.js:17:60)
    at build (/home/jozef-piecyk-10clouds/.config/yarn/global/node_modules/@netlify/build/src/core/main.js:53:35)
```

**- If the current behavior is a bug, please provide the steps to reproduce.**
1. Run `netlify build --dry` when the current user is not the owner of npm global files

**- What is the expected behavior?**

User is warned that the error is due to lack of permission.

**- Local Environment Information**
netlify version --verbose
Linux Mint 19 Cinnamon

**Comment by ehmicky** [comment]
Hi @masives,

Could you please print the output of `netlify version --verbose`. Thanks!

**Comment by masives** [comment]
oh, sorry about that 
`netlify-cli/2.20.0 linux-x64 node-v10.21.0`

**Comment by ehmicky** [comment]
This problem might be fixed in the latest Netlify CLI (`2.58.0` at of now). 
Would it be possible to upgrade, and try again? Thanks!

**Comment by masives** [comment]
What's interesting is that it shows different version when using sudo
![image](https://user-images.githubusercontent.com/18038829/89150264-784a5180-d55e-11ea-89a3-76431b1ff652.png)

Anyways it looks like my local setup causes it so I'm closing the issue
